pkg/config/analysis/local/istiod_analyze.go

```diff
@@ -306,11 +306,11 @@ func (sa *IstiodAnalyzer) AddRunningKubeSource(c kubelib.Client) {
 }
 
 func isIstioConfigMap(obj any) bool {
-	cObj, ok := obj.(controllers.Object)
+	cObj, ok := obj.(*v1.ConfigMap)
 	if !ok {
 		return false
 	}
-	return strings.HasPrefix(cObj.GetName(), "istio")
+	return strings.HasPrefix(cObj.GetName(), "istio") && !leaderelection.IsIstioLocks(cObj.GetName())
```

Comment:
Check the annotation `LeaderElectionRecordAnnotationKey = "control-plane.alpha.kubernetes.io/leader"`

Reply:
And i am not sure whether the analyzer check all namespaces endpoint/configmap. If so we should also ignore endpoints with the leader annotation

Reply:
> Check the annotation LeaderElectionRecordAnnotationKey = "control-plane.alpha.kubernetes.io/leader"

Fixed.

> And i am not sure whether the analyzer check all namespaces endpoint/configmap. If so we should also ignore endpoints with the leader annotation

Thanks for the reminder. Currently, no analyzer is checking the endpoint. I think we can add the filter logic once it is required by the analyzers.